src/ios/CDVCamera.m

```diff
@@ -360,7 +361,22 @@ - (NSData*)processImage:(UIImage*)image info:(NSDictionary*)info options:(CDVPic
         {
             if ((options.allowsEditing == NO) && (options.targetSize.width <= 0) && (options.targetSize.height <= 0) && (options.correctOrientation == NO) && (([options.quality integerValue] == 100) || (options.sourceType != UIImagePickerControllerSourceTypeCamera))){
                 // use image unedited as requested , don't resize
-                data = UIImageJPEGRepresentation(image, 1.0);
+                  if(options.sourceType != UIImagePickerControllerSourceTypeCamera){
+                    NSURL *url = info[UIImagePickerControllerReferenceURL];
+                    PHFetchResult *result = [PHAsset fetchAssetsWithALAssetURLs:@[url] options:nil];
+                    PHAsset *asset = [result firstObject];
+                    if (asset) {
+                        PHImageManager *manager = [PHImageManager defaultManager];
+                        PHImageRequestOptions *option = [PHImageRequestOptions alloc];
+                        option.synchronous = true;
+                        [manager requestImageDataForAsset:asset options:option resultHandler:^(NSData *imageData, NSString *dataUTI, UIImageOrientation orientation, NSDictionary *info) {
+                              data = imageData;
```

Comment:
This is the error that @alsorokin pointed out. Inherently this whole PR has to be re-worked since here on this line, this an async block (a callback) that resides in a synchronous function. For this to work, the processImage function has to be async, and changing that is not trivial.

Reply:
@shazron 
Its default functionality from `PHFetchResult` lib to return data from block to function, so it must be synchronous operation.
Let me know if you have other reference or solution to solved this bug.